jstp/src/main/java/com/metarhia/jstp/connection/AbstractSocket.java

```diff
@@ -10,34 +13,97 @@
    */
   boolean connect();
 
+  /**
+   * Sends the message
+   *
+   * @param message message
+   */
   void send(String message);
 
+  /**
+   * Pauses the connection
+   */
   void pause();
 
+  /**
+   * Resumes the connection
+   */
   void resume();
 
+  /**
+   * Closes the connection
+   *
+   * @param forced if the connection should be closed forced
+   */
   void close(boolean forced);
 
+  /**
+   * Clears the message queue
+   */
   void clearQueue();
 
+  /**
+   * Gets message queue size
+   *
+   * @return message queue size
+   */
   int getQueueSize();
 
+  /**
+   * Sets socket listener
+   *
+   * @param listener socket listener
+   */
   void setSocketListener(AbstractSocketListener listener);
 
+  /**
+   * Checks if socket is connected
+   *
+   * @return is socket connected
+   */
   boolean isConnected();
 
+  /**
+   * Checks if socket is closed
+   *
+   * @return is socket closed
+   */
   boolean isClosed();
 
+  /**
+   * Checks if connection is running
+   *
+   * @return is connection running
+   */
   boolean isRunning();
 
+  /**
+   * Abstract socket listener for connection events
+   */
   interface AbstractSocketListener {
 
+    /**
+     * Emitted when connection gets established
+     */
     void onConnected();
 
+    /**
+     * Emitted when message is received
+     *
+     * @param message message received
+     */
     void onMessageReceived(JSObject message);
 
+    /**
+     * Emitted when connection gets closed
+     */
     void onConnectionClosed();
 
+    /**
+     * Emitted when exception is thrown
```

Comment:
Called when error happened in the transport with possibly not null exception parameter.